check_stability.py

```diff
@@ -283,6 +286,17 @@ def wptrunner_args(self, root):
             "test_types": ["testharness", "reftest"]
         }
 
+    def extra_setup(self):
+        # https://bugs.chromium.org/p/chromium/issues/detail?id=713947
+        logger.debug("DBUS_SESSION_BUS_ADDRESS %s" % os.environ.get("DBUS_SESSION_BUS_ADDRESS"))
+        if "DBUS_SESSION_BUS_ADDRESS" not in os.environ:
+            if find_executable("dbus-launch"):
+                logger.debug("Attempting to start dbus")
+                logger.debug(subprocess.check_output(["dbus-launch"]))
+            else:
+                logger.debug("Setting DBUS_SESSION_BUS_ADDRESS to a placeholder value")
+                os.environ["DBUS_SESSION_BUS_ADDRESS"] = "/dev/null"
```

Comment:
My research is complete. The short version: let's fail the build instead of setting the variable to `/dev/null`.

The manual page for `dbus-launch` reads:

> If `DBUS_SESSION_BUS_ADDRESS` is not set for a process that tries to use
> D-Bus, by default the process will attempt to invoke `dbus-launch` with the
> `--autolaunch` option to start up a new session bus or find the existing bus
> address on the X display or in a file in `~/.dbus/session-bus/`

Which means that setting the variable to *any* value will have the intended result: stopping glib from attempting to start a session. However, setting it to `/dev/null` will effect the way Chrome interacts with the system. This is unlikely to be a problem today, but I'm concerned about future scenarios where tests are being written for the Web Bluetooth API (for example). I'd hate to trip anyone up just because they assumed that this low-level integration was valid.

The threat may seem kind of protracted, but I think it's fair to be conservative (call it paranoid if you like) here. That's because in reality, we have no use-case for this branch. [The latest job log](https://travis-ci.org/w3c/web-platform-tests/jobs/224408572#L416) demonstrates that TravisCI (i.e. the only environment that requires this workaround) offers the `dbus-launch` executable:

>     DEBUG:check_stability:DBUS_SESSION_BUS_ADDRESS None
>     DEBUG:check_stability:Attempting to start dbus
>     DEBUG:check_stability:DBUS_SESSION_BUS_ADDRESS=unix:abstract=/tmp/dbus-R7UYTMZ60G,guid=5b04beaf64b61ac2fe15c33858fa3141
>     DBUS_SESSION_BUS_PID=5839
>     DBUS_SESSION_BUS_WINDOWID=2097153

So the branch in question effectively states: "Do something potentially invalid in a case that we don't currently need to support." We *could* just remove that branch, but I think it would be more responsible to report a meaningful error for this case, just in case this script winds up getting ported to some new environment (FreeBSD on Docker...?). Something like, "Due to a bug in glib, a valid DBUS session address must be defined before starting Chrome."

In effect, the current branch silently implements a workaround for that future case, but it's a workaround that might have deeper ramifications. Reporting an error would give the future caller the knowledge they need to implement a version of this workaround that is suited to their environment (even including `DBUS_SESSION_BUS_ADDRESS=/dev/null`).